src/main/scala/com/microsoft/hyperspace/index/IndexLogManager.scala

```diff
@@ -111,13 +111,16 @@ class IndexLogManagerImpl(indexPath: Path) extends IndexLogManager with Logging
   }
 
   override def createLatestStableLog(id: Int): Boolean = {
-    // TODO: make sure log with the id has a stable state.
-    try {
-      FileUtil.copy(fs, pathFromId(id), fs, latestStablePath, false, new Configuration)
-    } catch {
-      case ex: Exception =>
-        logError(s"Failed to create the latest stable log with id = '$id'", ex)
-        false
+    getLog(id) match {
+      case Some(logEntry) if Constants.STABLE_STATES.contains(logEntry.state) =>
+        Try(FileUtil.copy(fs, pathFromId(id), fs, latestStablePath, false, new Configuration))
+          match {
+            case Success(v) => v
+            case Failure(e) =>
+              logError(s"Failed to create the latest stable log with id = '$id'", e)
+               false
+          }
+      case _ => false
```

Comment:
I think we need to catch two cases: 1) non stable state log entry and 2) `getLog` returned `None`. Can you also add a logging for each case?

Reply:
ok